exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/rm/DefaultResourceManager.java

```diff
@@ -73,13 +75,20 @@ public void setCost(double cost) {
     @Override
     public void admit() {
       // No queueing by default
+      foreman.moveToState(UserBitShared.QueryResult.QueryState.STARTING, null);
```

Comment:
Would REALLY prefer that the resource manager not depend on the Foreman; makes unit testing impossible. The Foreman should depend on this class, but not the other way around.